js_modules/dagster-ui/packages/ui-core/src/asset-graph/sidebar/Sidebar.tsx

```diff
@@ -434,3 +353,147 @@ export const AssetGraphExplorerSidebar = React.memo(
     );
   },
 );
+
+function buildRenderedNodes(nodes: {[assetId: string]: GraphNode}, openNodes: Set<string>) {
+  // Map of Groups -> Assets
+  const groupNodes: Record<string, {groupName: string; assets: GraphNode[]}> = {};
+
+  let groupsCount = 0;
+  Object.values(nodes).forEach((node) => {
+    const groupName = node.definition.groupName || 'default';
+    const groupId = groupIdForNode(node);
+
+    if (!groupNodes[groupId]) {
+      groupsCount += 1;
+    }
+    groupNodes[groupId] = groupNodes[groupId] || {
+      groupName,
+      assets: [],
+    };
+    groupNodes[groupId].assets.push(node);
+  });
+
+  const renderGroupsLayer = groupsCount > 1;
+  const folderNodes: FolderNodeType[] = [];
+
+  Object.entries(groupNodes)
+    .sort(([_1, a], [_2, b]) => COLLATOR.compare(a.groupName, b.groupName))
+    .forEach(([id, groupNode]) => {
+      if (renderGroupsLayer) {
+        folderNodes.push({groupNode, id, level: 1});
+      }
+      if (openNodes.has(id) || !renderGroupsLayer) {
+        groupNode.assets
+          .sort((a, b) => COLLATOR.compare(a.id, b.id))
+          .forEach((assetNode) => {
+            folderNodes.push({
+              id: assetNode.id,
+              path: [groupNode.groupName, tokenForAssetKey(assetNode.assetKey)].join(':'),
+              level: renderGroupsLayer ? 2 : 1,
+            });
+          });
+      }
+    });
+
+  return folderNodes;
+}
+
+function buildRenderedNodesWithCodeLocations(
```

Comment:
This function is unmodified from the old version, just pulled it into it's own helper so I could write a new version.

I made a new version instead of if/elsing a bunch of things in here because removing the code location layer actually makes the logic a lot simpler.